### PR TITLE
Build Matrix for local example (macOS and Ubuntu)

### LINKED ISSUE #5535: Add macOS and CentOS local_example tests
#### Feature Description

The [local_example test](https://github.com/vitessio/vitess/blob/master/.github/workflows/local_example.yml) currently tests the latest Ubuntu LTS (18.04). We should add versions of this for CentOS and macOS, preferably using matrix builds.

This will help catch if there is something specific introduced which breaks other OSes.. which has happened in the past.

### PR DESCRIPTION
**Pull request by J0**
Fix #5535

Signed-off-by: Morgan Tocker <tocker@gmail.com>

### CODE REVIEW

**Comment by morgo** [comment]
For this to merge, we will need to change the required test from local_example to local_example on Ubuntu.

File: .github/workflows/local_example.yml [hunk 1]
```diff
@@ -3,8 +3,12 @@ on: [push, pull_request]
 jobs:
 
   build:
-    name: Local Example Test
-    runs-on: ubuntu-latest
+    name: Local Example for ${{ matrix.os }}
```
Comment (enisoc) [inline .github/workflows/local_example.yml]:
```suggestion
    name: Local Example on ${{ matrix.os }}
```

**Comment by morgo** [comment]
I think there is some flakyness on macOS with bringing up the tablets. I will take a look at it before we merge this.

**Comment by morgo** [comment]
I could not reproduce the flakyness locally, but I have changed mac os to bind to `localhost`, which seems to be working.

**Pull request merged**